sw/device/tests/kmac_error_conditions_test.c

```diff
@@ -0,0 +1,447 @@
+
+    // Reset FSM by setting `err_processed` bit
+    TRY(dif_kmac_reset(&kmac, &kmac_operation_state));
+
+    // At this point, we expect that there are no remaining interrupts.
+    dif_kmac_irq_state_snapshot_t intr_snapshot;
+    TRY(dif_kmac_irq_get_state(&kmac, &intr_snapshot));
+    CHECK(intr_snapshot == 0, "INTR_STATE is non-zero after timeout clean-up.");
+
+    bool is_kmac_locked;
+    TRY(dif_kmac_config_is_locked(&kmac, &is_kmac_locked));
+    CHECK(!is_kmac_locked, "KMAC still locked after timeout clean-up.");
+  }
+
+  return OK_STATUS();
+}
+
+/**
+ * ErrIncorrectEntropyMode.
+ *
+ * This test checks whether the ErrIncorrectEntropyMode error code is
+ * returned by the hardware when using a wrong entropy mode configuration.
+ *
+ * @return OK or error.
+ */
+status_t test_err_incorrect_entropy_mode(void) {
+  LOG_INFO("Testing ErrIncorrectEntropyMode error.");
+
+  // Re-init KMAC for the test.
+  TRY(dif_kmac_init(mmio_region_from_addr(TOP_EARLGREY_KMAC_BASE_ADDR), &kmac));
+
+  // Write configuration register.
+  uint32_t cfg_reg = 0;
+  // Provide invalid entropy mode (other than kDifKmacEntropyModeSoftware or
+  // kDifKmacEntropyModeEdn)
+  cfg_reg = bitfield_field32_write(cfg_reg,
+                                   KMAC_CFG_SHADOWED_ENTROPY_MODE_FIELD, 0xf);
+  // Set remaining config to a valid one.
+  cfg_reg = bitfield_bit32_write(cfg_reg, KMAC_CFG_SHADOWED_MSG_ENDIANNESS_BIT,
+                                 kDifToggleDisabled);
+  cfg_reg = bitfield_bit32_write(
+      cfg_reg, KMAC_CFG_SHADOWED_STATE_ENDIANNESS_BIT, kDifToggleDisabled);
+  cfg_reg = bitfield_bit32_write(
+      cfg_reg, KMAC_CFG_SHADOWED_ENTROPY_FAST_PROCESS_BIT, kDifToggleDisabled);
+  cfg_reg = bitfield_bit32_write(cfg_reg, KMAC_CFG_SHADOWED_SIDELOAD_BIT,
+                                 kDifToggleDisabled);
+  cfg_reg =
+      bitfield_bit32_write(cfg_reg, KMAC_CFG_SHADOWED_ENTROPY_READY_BIT, true);
+  cfg_reg = bitfield_bit32_write(cfg_reg, KMAC_CFG_SHADOWED_MSG_MASK_BIT,
+                                 kDifToggleEnabled);
+  mmio_region_write32_shadowed(kmac.base_addr, KMAC_CFG_SHADOWED_REG_OFFSET,
+                               cfg_reg);
+
+  // Read out error information.
+  dif_kmac_error_t err;
+  uint32_t info;
+  TRY(dif_kmac_get_error(&kmac, &err, &info));
+  CHECK(err == kDifErrorEntropyModeIncorrect,
+        "Error other than kDifErrorEntropyModeIncorrect occurred.");
+
+  // Reset the KMAC FSM.
+  dif_kmac_operation_state_t kmac_operation_state;
+  TRY(dif_kmac_reset(&kmac, &kmac_operation_state));
+
+  return OK_STATUS();
+}
+
+/**
+ * ErrorSoftwareCommandSequence.
+ *
+ * This test checks whether the kDifErrorSoftwareCommandSequence error code is
+ * returned by the hardware when sending software commands in a wrong order to
+ * the KMAC block. Note that this test is not exhaustive, i.e., the test does
+ * not trying to reach the ErrorSoftwareCommandSequence error state from each
+ * other state.
```

Comment:
Thanks for the comment, I think this is okay but maybe it would be good for someone from the SiVal team to confirm @luismarques , @engdoreis ?